protos/sources/vega/api/v1/core.proto

```diff
@@ -343,3 +348,84 @@ message LastBlockHeightResponse {
   // the network chain id from which the block comes from
   string chain_id = 8;
 }
+
+// A request to retrieve the spam statistics of a party for the given epoch
+message GetSpamStatisticsRequest {
+  // Id for the party whose statistics are requested
+  string party_id = 1;
+}
+
+// The statistics for a given spam policy
+message SpamStatistic {
+  // The current transaction count received from the party during this epoch for this policy
+  uint64 count_for_epoch = 1;
+  // The maximum number of transactions allowed for this policy in an epoch
+  uint64 max_for_epoch = 2;
+  // If blocked the timestamp when the party will be unblocked as RFC3339Nano
+  optional string banned_until = 4;
+}
+
+// The voting statistics by proposal for a given party for the current epoch
+message VoteSpamStatistics {
+  // List of statistics for proposals voted on by the party
+  repeated VoteSpamStatistic statistics = 1;
+  // If blocked the timestamp when the party will be unblocked as RFC3339Nano
+  optional string banned_until = 2;
+}
+
+// The vote statistics for the voting spam policies
+// which are calculated as a ratio of the total votes
+// that have been rejected.
+message VoteSpamStatistic {
+  string proposal = 1;
+  // The current vote count received from the party for the given proposal during this epoch
+  uint64 count_for_epoch = 2;
+  // The maximum number of votes per proposal allowed in an epoch
+  uint64 max_for_epoch = 3;
```

Comment:
Either I understand this property documentation the wrong way, either it shouldn't live in this payload. `VoteSpamStatistic` seems to be related to a "single" proposal. However `max_for_epoch` seems to be a threshold set for all proposals. Wouldn't this make the information redundant ?

Reply:
it's max_per_proposal, but its definition is global and it shouldn't be here.